raiden/network/proxies/token_network.py

```diff
@@ -899,6 +903,212 @@ def _check_why_deposit_failed(
 
         return error_type, msg
 
+    def _withdraw_preconditions(
+        self,
+        channel_identifier: ChannelID,
+        total_withdraw: WithdrawAmount,
+        previous_total_withdraw: WithdrawAmount,
+        partner: Address,
+        block_identifier: BlockSpecification,
+    ) -> None:
+        if not self.client.can_query_state_for_block(block_identifier):
+            raise NoStateForBlockIdentifier()
+
+        self._check_for_outdated_channel(
+            participant1=self.node_address,
+            participant2=partner,
+            block_identifier=block_identifier,
+            channel_identifier=channel_identifier,
+        )
+
+        amount_to_withdraw = total_withdraw - previous_total_withdraw
+
+        if total_withdraw <= previous_total_withdraw:
+            msg = (
+                f"Current total withdraw ({previous_total_withdraw}) is already larger "
+                f"than the requested total withdraw amount ({total_withdraw})"
+            )
+            raise DepositMismatch(msg)
+
+        if amount_to_withdraw <= 0:
```

Comment:
I believe this condition can never be met / will always be caught in [line 926](https://github.com/raiden-network/raiden/pull/4130/files#diff-d02574a7a7eca826412780e80a6326a3R926)?

```python
amount_to_withdraw = total_withdraw - previous_total_withdraw
if amount_to_withdraw <= 0:
  assert total_withdraw <= previous_total_withdraw
```

Reply:
Done